src/containers/role-management/role-create.tsx

```
@@ -0,0 +1,294 @@
+    const groups = Constants.PERMISSIONS;
+    const { permissions: selectedPermissions, name } = this.state;
+
+    const { featureFlags } = this.context;
+    let isUserMgmtDisabled = false;
+    const filteredPermissions = { ...Constants.HUMAN_PERMISSIONS };
+    if (featureFlags) {
+      isUserMgmtDisabled = featureFlags.external_authentication;
+    }
+    if (isUserMgmtDisabled) {
+      Constants.USER_GROUP_MGMT_PERMISSIONS.forEach((perm) => {
+        if (perm in filteredPermissions) {
+          delete filteredPermissions[perm];
+        }
+      });
+    }
+
+    const notAuthorised = !this.context.user || this.context.user.is_anonymous;
+    const breadcrumbs = [
+      { url: Paths.roleList, name: t`Roles` },
+      { name: t`Create new role` },
+    ];
+    const title = t`Create new role`;
+
+    return (
+      <React.Fragment>
+        <BaseHeader
+          breadcrumbs={<Breadcrumbs links={breadcrumbs}></Breadcrumbs>}
+          title={title}
+        ></BaseHeader>
+        {notAuthorised ? (
+          <EmptyStateUnauthorized />
+        ) : (
+          <Main>
+            <section className='body'>
+              <div>
+                <div style={{ paddingBottom: '8px', paddingTop: '16px' }}>
+                  <Title headingLevel='h2'>Details</Title>
+                </div>
+                <FormGroup
+                  isRequired={true}
+                  key='name'
+                  fieldId='name'
+                  label={t`Name`}
+                  helperTextInvalid={t`Role name must be unique.`}
+                  validated={this.state.nameError ? 'error' : null}
+                >
+                  <TextInput
+                    id='role_name'
+                    value={this.state.name}
+                    onChange={(value) => {
+                      this.setState({ name: value, nameError: false });
+                    }}
+                    type='text'
+                    validated={this.state.nameError ? 'error' : null}
```

We need to do some extra validation here. If the role name doesn't start with `galaxy.` we need to throw an error telling the user that all custom roles must start with the prefix `galaxy.`.

---

API returns something like this `{name: 'This field must be unique'}` so it can be cleaned to be nicer. But maybe it should be a followup PR?